fe/fe-core/src/main/java/com/starrocks/warehouse/WarehouseIdleChecker.java

```diff
@@ -0,0 +1,139 @@
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class WarehouseIdleChecker extends FrontendDaemon {
+    private static final Logger LOG = LogManager.getLogger(WarehouseIdleChecker.class);
+
+    private static final Map<Long, AtomicLong> RUNNING_SQL_COUNT = new ConcurrentHashMap<>();
+
+    private static final Map<Long, Long> LAST_FINISHED_JOB_TIME = new ConcurrentHashMap<>();
+
+    private final Map<Long, Long> warehouseIdleTime = new ConcurrentHashMap<>();
+
+    public WarehouseIdleChecker() {
+        super("WarehouseIdleChecker", Config.warehouse_idle_check_interval_seconds * 1000);
+    }
+
+    @Override
+    protected void runAfterCatalogReady() {
+        if (!Config.warehouse_idle_check_enable) {
+            return;
+        }
+
+        if (getInterval() != Config.warehouse_idle_check_interval_seconds * 1000) {
+            setInterval(Config.warehouse_idle_check_interval_seconds * 1000);
+        }
+
+        List<Long> warehouseIds = GlobalStateMgr.getCurrentState().getWarehouseMgr().getAllWarehouseIds();
+        Map<Long, Long> runningStreamLoadCnt = GlobalStateMgr.getCurrentState().getStreamLoadMgr().getRunningTaskCount();
+        Map<Long, Long> runningLoadCnt = GlobalStateMgr.getCurrentState().getLoadMgr()
+                .getRunningLoadCount();
+        Map<Long, Long> runningRoutineLoadCnt = GlobalStateMgr.getCurrentState().getRoutineLoadMgr()
+                .getRunningRoutingLoadCount();
+        Map<Long, Long> runningBackupRestoreCnt = GlobalStateMgr.getCurrentState().getBackupHandler()
+                .getRunningBackupRestoreCount();
+        Map<Long, Long> runningAlterJobCnt = GlobalStateMgr.getCurrentState().getAlterJobMgr().getRunningAlterJobCount();
+        Map<Long, Long> runningTaskCnt = GlobalStateMgr.getCurrentState().getTaskManager().getTaskRunScheduler()
+                .getAllRunnableTaskCount();
+
+        for (long wId : warehouseIds) {
+            long lastFinishedJobTime = LAST_FINISHED_JOB_TIME.getOrDefault(wId, -1L);
+            long runningJobCnt = 0;
+            runningJobCnt += getRunningSQLCount(wId).get();
+            runningJobCnt += runningStreamLoadCnt.getOrDefault(wId, 0L);
+            runningJobCnt += runningLoadCnt.getOrDefault(wId, 0L);
+            runningJobCnt += runningRoutineLoadCnt.getOrDefault(wId, 0L);
+            runningJobCnt += runningBackupRestoreCnt.getOrDefault(wId, 0L);
+            runningJobCnt += runningAlterJobCnt.getOrDefault(wId, 0L);
+            runningJobCnt += runningTaskCnt.getOrDefault(wId, 0L);
+
+            if (runningJobCnt == 0
+                    && lastFinishedJobTime <
+                        System.currentTimeMillis() - Config.warehouse_idle_check_interval_seconds * 2000) {
+                warehouseIdleTime.putIfAbsent(wId, System.currentTimeMillis());
+                LOG.info("warehouse: {} is idle, idle start time: {}",
```

Comment:
once the warehouse is idle, will this message be printed every interval, will it be too chatty?

Reply:
Maybe this is useful for users to close the unused warehouse.